backend/src/main/java/com/benine/backend/AutoPresetCreator.java

```diff
@@ -27,17 +25,28 @@
    * @throws CameraConnectionException when camera cannot be reached.
    * @throws InterruptedException when interupted while waiting for cam to move.
    * @throws TimeoutException if the camera is moving too slow or not at all.
+   * @throws CameraBusyException if the camera is busy.
+   * @throws IOException if exception occurs when creating the preset image.
+   * @throws StreamNotAvailableException if a camera stream cannot be reached.
    */
   public Collection<Preset> createPresets(IPCamera cam)
-          throws CameraConnectionException, InterruptedException, TimeoutException, IOException, StreamNotAvailableException {
+          throws CameraConnectionException, CameraBusyException, InterruptedException,
+          TimeoutException, IOException, StreamNotAvailableException {
+    if (cam.isBusy()) {
+      throw new CameraBusyException("The camera is busy.", cam.getId());
+    }
+    cam.setBusy(true);
     Position camStartPos = cam.getPosition();
 
     ArrayList<Preset> presets = new ArrayList<Preset>();
     for (ZoomPosition pos : generatePositions(cam)) {
+      cam.setBusy(false);
```

Comment:
The problem with this is we would need to make a distinction between interacting with the camera from the AutoPresetCreator and the handlers. Which would imply moving checking if the camera is busy to to the Httphandlers (and other places in the system) where the camera's might be controlled. This clashes with the single responsibility system. It's the camera's responsibility to not allow other commands to be executed when its busy. Also if we extend the busy in the future we could also use it to enforce the single command per camera per 130ms rule using this. Then it would be usefull to check here if the camera is busy, if it is then wait and try again after a little while.